tests/Spec/Nri/Ui/ClickableText.elm

```diff
@@ -1,72 +1,411 @@
-module Spec.Nri.Ui.ClickableText exposing (spec)
+module Spec.Nri.Ui.ClickableText exposing (..)
 
 import Accessibility.Aria as Aria
-import Html.Styled exposing (Html, toUnstyled)
+import Accessibility.Role as Role
+import Html.Attributes as Attributes
+import Html.Styled exposing (..)
+import Nri.Test.KeyboardHelpers.V1 as KeyboardHelpers
+import Nri.Test.MouseHelpers.V1 as MouseHelpers
 import Nri.Ui.ClickableText.V3 as ClickableText
+import Nri.Ui.UiIcon.V1 as UiIcon
 import ProgramTest exposing (..)
 import Spec.Helpers exposing (expectFailure)
 import Test exposing (..)
-import Test.Html.Selector exposing (..)
+import Test.Html.Event as Event
+import Test.Html.Query as Query
+import Test.Html.Selector as Selector exposing (..)
 
 
 spec : Test
 spec =
     describe "Nri.Ui.ClickableText.V3"
-        [ describe "helpfullyDisabledClickableText" helpfullyDisabledClickableText
+        [ describe "elements" elementTests
+        , describe "attributes" attributeTests
+        , describe "icon accessibility" iconAccessibilityTests
+        , describe "disabled behavior and attributes" disabledStateTests
         ]
 
 
-helpfullyDisabledClickableText : List Test
-helpfullyDisabledClickableText =
-    [ test "does not have `aria-disabled=\"true\" when not disabled" <|
+type Type_
+    = Button
+    | Link
+
+
+elementTests : List Test
+elementTests =
+    [ testAsButton
+    , testAsAnchor
+    , testIconAsSvg
+    , testRightIconAsSvg
+    , testLinkExternalIconAsSvg
+    ]
+
+
+testAsButton : Test
+testAsButton =
+    test "the `button` type renders as a button element" <|
```

Comment:
I think we usually nest describes and tests directly. Is there a reason to flatten the specs and organizing them as proposed?

I checked on our elm style guide and there is nothing against but it seems like a deviation from what we usually do, so I wonder the motivations in case I am missing something.

Reply:
no specific reason, I just thought it would look cleaner, but don't feel strongly. Updated!